GitUI/HelperDialogs/FormRemoteProcess.cs

```diff
@@ -68,6 +68,7 @@ protected override void BeforeProcessStart()
         {
             _restart = false;
             Plink = GitSshHelpers.IsPlink;
+            PasswordPanel.Visible = true;
```

Comment:
This means the new textbox is always shown, which isn't good. I use OpenSSH and I don't need or like to see this textbox.

Reply:
I explained it above, but there's no reliable way to detect if and when a password is being asked for, at least not without integrating with the SSH client a heck of a lot tighter than simple stdin/stdout piping. The issue stems from the fact that neither SSH nor Git report a successful login. A request from a password is indistinguishable from a successful public key login followed by a commit message that mentions "password" in its text.

This is why something as invasive as a popup is an absolute no-go. You can't do anything that would block, or even appear to block, the current operation.

The current placement is as uninvasive as possible. A good comparison would be the "Branches" and "Filters" boxes at the top of the main window. I don't need or like to see them, but I don't demand they be removed, because their functionality may be necessary to someone.